packages/fxa-settings/src/components/FormPasswordWithBalloons/index.tsx

```diff
@@ -0,0 +1,207 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import { UseFormMethods } from 'react-hook-form';
+import InputPassword from '../InputPassword';
+import PasswordValidator from '../../lib/password-validator';
+import { logViewEvent, settingsViewName } from '../../lib/metrics';
+import PasswordStrengthBalloon from '../PasswordStrengthBalloon';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { ReactComponent as IconKey } from './icon-key-grey-50.svg';
+import { useFtlMsgResolver } from '../../models';
+
+type FormPasswordWithBalloonsProps = {
+  formState: UseFormMethods['formState'];
+  errors: UseFormMethods['errors'];
+  onSubmit: any; // TODO: fix this type
+  trigger: UseFormMethods['trigger'];
+  register: UseFormMethods['register'];
+  getValues: UseFormMethods['getValues'];
+  email: string;
+  onFocusMetricsEvent?: string;
+  loading: boolean;
+  newPasswordLabel?: string;
+  confirmPasswordLabel?: string;
+  submitButtonText?: string;
+  children?: React.ReactNode;
+  showConfirmPwdBalloon?: boolean;
+};
+
+export const FormPasswordWithBalloons = ({
+  formState,
+  errors,
+  onSubmit,
+  email,
+  trigger,
+  register,
+  getValues,
+  onFocusMetricsEvent,
+  loading,
+  newPasswordLabel,
+  confirmPasswordLabel,
+  submitButtonText,
+  children,
+  showConfirmPwdBalloon,
+}: FormPasswordWithBalloonsProps) => {
+  const passwordValidator = new PasswordValidator(email);
+  const [hasNewPwdFocused, setHasNewPwdFocused] = useState(false);
+  const [isNewPwdBalloonVisible, setIsNewPwdBalloonVisible] = useState(false);
+  const [isConfirmPwdBalloonVisible, setIsConfirmPwdBalloonVisible] =
+    useState(false);
+  const [passwordMatchErrorText, setPasswordMatchErrorText] = useState<
+    string | undefined
+  >(undefined);
+  const ftlMsgResolver = useFtlMsgResolver();
+
+  const defaultButtonText = ftlMsgResolver.getMsg(
+    'form-reset-password-with-balloon-submit-button',
+    'Reset password'
+  );
+
+  const onNewPwdFocus = () => {
+    setIsNewPwdBalloonVisible(true);
+    if (!hasNewPwdFocused && onFocusMetricsEvent) {
+      logViewEvent(settingsViewName, onFocusMetricsEvent);
+      setHasNewPwdFocused(true);
+    }
+  };
+
+  const onNewPwdBlur = () => {
+    setIsNewPwdBalloonVisible(false);
+  };
+
+  const onConfirmPwdFocus = () => {
+    setIsConfirmPwdBalloonVisible(true);
+    console.log('triggered');
```

Comment:
Q: do we still need the logging?

Reply:
We do not! Thanks for catching this straggler.